classes/gateways/class.pmprogateway_stripe.php

```diff
@@ -3361,17 +3403,22 @@ static function stripe_connect_save_options() {
 			|| ! isset( $_REQUEST['pmpro_stripe_publishable_key'] )
 			|| ! isset( $_REQUEST['pmpro_stripe_user_id'] )
 			|| ! isset( $_REQUEST['pmpro_stripe_access_token'] )
-			|| ! isset( $_REQUEST['pmpro_stripe_access_token_test'] )
-			|| ! isset( $_REQUEST['pmpro_stripe_publishable_key_test'] )
 		) {
 			$error = __( 'Invalid response from the Stripe Connect server.', 'paid-memberships-pro' );
 		} else {
 			// Update keys.
-			pmpro_setOption( 'stripe_connect_user_id', $_REQUEST['pmpro_stripe_user_id'] );
-			pmpro_setOption( 'live_stripe_connect_secretkey', $_REQUEST['pmpro_stripe_access_token'] );
-			pmpro_setOption( 'test_stripe_connect_secretkey', $_REQUEST['pmpro_stripe_access_token_test'] );
-			pmpro_setOption( 'live_stripe_connect_publishablekey', $_REQUEST['pmpro_stripe_publishable_key'] );
-			pmpro_setOption( 'test_stripe_connect_publishablekey', $_REQUEST['pmpro_stripe_publishable_key_test'] );
+			if ( $_REQUEST['pmpro_stripe_connected_environment'] === 'live' ) {
+				// Update live keys.
+				pmpro_setOption( 'live_stripe_connect_user_id', $_REQUEST['pmpro_stripe_user_id'] );
+				pmpro_setOption( 'live_stripe_connect_secretkey', $_REQUEST['pmpro_stripe_access_token'] );
+				pmpro_setOption( 'live_stripe_connect_publishablekey', $_REQUEST['pmpro_stripe_publishable_key'] );
+			} else {
+				// Update test keys.
+				pmpro_setOption( 'test_stripe_connect_user_id', $_REQUEST['pmpro_stripe_user_id'] );
+				pmpro_setOption( 'test_stripe_connect_secretkey', $_REQUEST['pmpro_stripe_access_token'] );
+				pmpro_setOption( 'test_stripe_connect_publishablekey', $_REQUEST['pmpro_stripe_publishable_key'] );
```

Comment:
```suggestion
				pmpro_setOption( 'live_stripe_connect_user_id', sanitize_text_field( $_REQUEST['pmpro_stripe_user_id'] ) );
				pmpro_setOption( 'live_stripe_connect_secretkey', sanitize_text_field( $_REQUEST['pmpro_stripe_access_token'] ) );
				pmpro_setOption( 'live_stripe_connect_publishablekey', sanitize_text_field( $_REQUEST['pmpro_stripe_publishable_key'] ) );
			} else {
				// Update test keys.
				pmpro_setOption( 'test_stripe_connect_user_id', sanitize_text_field( $_REQUEST['pmpro_stripe_user_id'] ) );
				pmpro_setOption( 'test_stripe_connect_secretkey', sanitize_text_field( $_REQUEST['pmpro_stripe_access_token'] ) );
				pmpro_setOption( 'test_stripe_connect_publishablekey', sanitize_text_field( $_REQUEST['pmpro_stripe_publishable_key'] ) );
```

Reply:
pmpro_setOption defaults to using sanitize_text_field. So not needed.
https://github.com/strangerstudios/paid-memberships-pro/blob/dev/includes/functions.php#L87

Sometimes it's worth being clear for the plugin team reviewers. We can add a comment RE that at the top